src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -256,6 +257,12 @@ private boolean launchNode(SchedulerDriver driver, Offer offer,
     //executorName is to distinguish different types of nodes
     //taskType is the type of task in mesos to launch on the node
     //taskName is a name chosen to identify the task in mesos and mesos-dns (if used)
```

Comment:
I think you can actually safely revert these changes. I don't think this is related, because if it enters this method, not only would it create an executor, but it would also call driver.launchTasks. Unless phantom tasks are getting created, I think you only need to make the changes for ProcessBuilder and the executors, but not here.